File: puzzles/python3/the-descent/the_descent.py
while True:
    highest_index = 0
    highest_height = -1

    # Read the heights of the mountains and determine the highest
    for i in range(8):
        mountain_height = int(input())
        
        # Check if this mountain is the highest so far
        if mountain_height > highest_height:
            highest_height = mountain_height
            highest_index = i

    # Output the index of the highest mountain to shoot
    print(highest_index)

File: puzzles/cpp/the-descent/the_descent.cpp
#include <iostream>
using namespace std;

int main() {
    // Game loop
    while (true) {
        int highestIndex = 0;
        int highestHeight = -1;

        // Read the heights of the mountains and determine the highest
        for (int i = 0; i < 8; i++) {
            int mountainHeight;
            cin >> mountainHeight;

            // Check if this mountain is the highest so far
            if (mountainHeight > highestHeight) {
                highestHeight = mountainHeight;
                highestIndex = i;
            }
        }

        // Output the index of the highest mountain to shoot
        cout << highestIndex << endl;
    }
}
